mechafil/power.py

```diff
@@ -66,8 +71,11 @@ def compute_day_qa_renewed_power(
     duration: int = None,
 ):
     qa_factor = compute_qa_factor(fil_plus_rate, fil_plus_m, duration_m, duration)
+    # day_renewed_power = (
+    #     qa_factor * renewal_rate_vec[-1:] * day_rb_scheduled_expire_power_vec[day_i]
+    # )
     day_renewed_power = (
-        qa_factor * renewal_rate_vec[-1:] * day_rb_scheduled_expire_power_vec[day_i]
+        qa_factor * renewal_rate_vec[day_i] * day_rb_scheduled_expire_power_vec[day_i]
```

Comment:
`fil_plus_rate` can be an array, right? In that case, `qa_factor` will also be an array and thus we should multiply this expression by `qa_factor[day]`.

Also, do not forget to update the the types in the function definition. I also realised and I forgot to add the return type in the def. Can you add it?

Reply:
addressed in a better way per your recommendations.